### PR TITLE
Moving HashPassword logic from AccountService

### PR DESCRIPTION
**Pull request by SerjKocherga**
HashPassword methods were moved from AccontService to PasswordHelper.

### CODE REVIEW

**Review by Hlib-mats — APPROVED** [review]

File: CharlieBackend.Business/Providers/HashPasswordProvider.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                Position = Next(0, _saltAlphabet.Length - 1);
+
+                //add the selected character to the object StringBuilder
+                sb.Append(_saltAlphabet[Position]);
+            }
+
+            return sb.ToString();
+        }
+
+        public static Int32 Next(Int32 minValue, Int32 maxValue)
```
Comment (whereismyroot) [inline CharlieBackend.Business/Providers/HashPasswordProvider.cs:31]: You don't need this method as public

File: CharlieBackend.Business/Providers/HashPasswordProvider.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+{
+    public static class HashPasswordProvider
+    {
+        private static readonly string _saltAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-01234567890";
+        private static readonly int _saltLen = 15;
+
+        public static string GenerateSalt()
+        {
+            //StringBuilder object with a predefined buffer size for the resulting string
+            StringBuilder sb = new StringBuilder(_saltLen - 1);
```
Comment (SnipeFrost) [inline CharlieBackend.Business/Providers/HashPasswordProvider.cs:15]: rename pls "sb" for more obvious name

File: CharlieBackend.Business/Providers/HashPasswordProvider.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Text;
+using System.Security.Cryptography;
+
+namespace CharlieBackend.Business.Providers
+{
+    public static class HashPasswordProvider
```
Comment (whereismyroot) [inline CharlieBackend.Business/Providers/HashPasswordProvider.cs:7]: please rename this to Helpers/PasswordHelper.cs since this class is static

**Review by BohdanBBG — APPROVED** [review]

**Review by whereismyroot — CHANGES_REQUESTED** [review]

File: CharlieBackend.Business/Services/AccountService.cs [hunk 4]
```diff
@@ -158,12 +151,12 @@ public async Task<Result<AccountDto>> ChangePasswordAsync(ChangeCurrentPasswordD
...
             {
-                string checkPassword = HashPassword(changePasswd.CurrentPassword, salt);
+                string checkPassword = HashPasswordProvider.HashPassword(changePasswd.CurrentPassword, salt);
 
                 if (user.Password == checkPassword)
                 {
-                    user.Salt = GenerateSalt();
-                    user.Password = HashPassword(changePasswd.NewPassword, user.Salt);
+                    user.Salt = HashPasswordProvider.GenerateSalt();
+                    user.Password = HashPasswordProvider.HashPassword(changePasswd.NewPassword, user.Salt);
```
Comment (SnipeFrost) [inline CharlieBackend.Business/Services/AccountService.cs:159]: why this property have name "changePasswd" ?

**Review by SnipeFrost — CHANGES_REQUESTED** [review]

**Review by SnipeFrost — APPROVED** [review]

**Review by sober607 — APPROVED** [review]

File: CharlieBackend.Business/Helpers/PasswordHelper.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        private static readonly string _saltAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-01234567890";
+        private static readonly int _saltLen = 15;
+
+        public static string GenerateSalt()
+        {
+            //StringBuilder object with a predefined buffer size for the resulting string
+            StringBuilder stringBuilder = new StringBuilder(_saltLen - 1);
+
+            //a variable for storing a random character position from the string Str
```
Comment (x1mx1x) [inline CharlieBackend.Business/Helpers/PasswordHelper.cs:17]: you should remove comments before merging

File: CharlieBackend.Business/Helpers/PasswordHelper.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+            return BitConverter.ToString(result).Replace("-", "").ToLower();
+        }
+
+        private static Int32 Next(Int32 minValue, Int32 maxValue)
+        {
+            using (RNGCryptoServiceProvider rng = new RNGCryptoServiceProvider())
+            {
+                byte[] uint32Buffer = new byte[4];
+                Int64 diff = maxValue - minValue;
```
Comment (x1mx1x) [inline CharlieBackend.Business/Helpers/PasswordHelper.cs:44]: why do you not use pseudonims?

**Review by x1mx1x — APPROVED** [review]

**Review by whereismyroot — APPROVED** [review]

**Pull request merged**